pkg/status/status_collector.go

```diff
@@ -274,6 +274,19 @@ func (d *statusCollector) getCNIChainingStatus() *models.CNIChainingStatus {
 	}
 }
 
+func (d *statusCollector) getBGPStatus(ctx context.Context) *models.BGPStatus {
```

Comment:
I just realized - we probably don't want to query the BGP status getter if BGP is disabled completely. My suggestion:
```
if !d.statusParams.DaemonConfig.EnableBGPControlPlane {
   return &models.BGPStatus{
      Mode: models.BGPStatusStateDisabled,
   }
}
```